src/System.IO.FileSystem.Watcher/src/System/IO/FileSystemWatcher.cs

```diff
@@ -105,8 +105,8 @@ public FileSystemWatcher(string path, string filter)
                 throw new ArgumentNullException(nameof(filter));
 
             // Early check for directory parameter so that an exception can be thrown as early as possible.
-            if (path.Length == 0 || !Directory.Exists(path))
-                throw new ArgumentException(SR.Format(SR.InvalidDirName, path), nameof(path));
+            if (!Directory.Exists(path))
```

Comment:
IMO the check for `path.Length == 0` should stay and output the current error message and after it the directory exists check with the more concrete error message.